app/channel.go

```diff
@@ -1198,3 +1198,39 @@ func GetPinnedPosts(channelId string) (*model.PostList, *model.AppError) {
 		return result.Data.(*model.PostList), nil
 	}
 }
+
+func GetOrCreateChannel(teamId, userId, channelNameOrId string) (*model.Channel, *model.AppError) {
```

Comment:
no functional changes here. more or less cut / pasted from _app/webhook.go_ so i could re-use it